### PR TITLE
(CDAP-3627) Update examples to use HttpContentConsumer for file upload

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Comment by awholegunch** [comment]
This will change the docs, as two of these files (FileSetService.java and UploadService.java) are guarded files; they are included in the docs and these changes will change the line numbers required.

**Comment by chtyim** [comment]
@awholegunch How to change the doc for the guarded files (besides adding new info about usage of HttpContentConsumer)?

File: cdap-examples/DataCleansing/src/main/java/co/cask/cdap/examples/datacleansing/DataCleansingService.java [hunk 1]
```diff
@@ -53,25 +60,49 @@ protected void configure() {
...
-        partitionOutput.addPartition();
+          @Override
+          public void onError(HttpServiceResponder responder, Throwable failureCause) {
+            Closeables.closeQuietly(channel);
+            try {
+              location.delete();
+            } catch (IOException e) {
+              LOG.warn("Failed to delete {}", location, e);
+            }
+            LOG.debug("Unable to write path '{}'", location);
```
Comment (anew) [inline cdap-examples/DataCleansing/src/main/java/co/cask/cdap/examples/datacleansing/DataCleansingService.java]: log the exception, too?

File: cdap-examples/FileSetExample/src/main/java/co/cask/cdap/examples/fileset/FileSetService.java [hunk 2]
```diff
@@ -87,28 +95,49 @@ public void read(HttpServiceRequest request, HttpServiceResponder responder,
...
+          @Override
+          public void onError(HttpServiceResponder responder, Throwable failureCause) {
+            Closeables.closeQuietly(channel);
+            try {
+              location.delete();
+            } catch (IOException e) {
+              LOG.warn("Failed to delete {}", location, e);
+            }
+            LOG.debug("Unable to write path '{}' in file set '{}'", filePath, set, failureCause);
+            responder.sendError(400, String.format("Unable to write path '%s' in file set '%s'", filePath, set));
```
Comment (anew) [inline cdap-examples/FileSetExample/src/main/java/co/cask/cdap/examples/fileset/FileSetService.java]: include the message of the failureCause?

File: cdap-examples/SportResults/src/main/java/co/cask/cdap/examples/sportresults/UploadService.java [hunk 3]
```diff
@@ -97,21 +105,47 @@ public void write(HttpServiceRequest request, HttpServiceResponder responder,
...
+
+          @Override
+          public void onError(HttpServiceResponder responder, Throwable failureCause) {
+            Closeables.closeQuietly(channel);
+            try {
+              partitionDir.delete(true);
+            } catch (IOException e) {
+              LOG.warn("Failed to delete partition directory '{}'", partitionDir, e);
+            }
+            responder.sendError(400, String.format("Unable to write path '%s'", output.getRelativePath()));
```
Comment (anew) [inline cdap-examples/SportResults/src/main/java/co/cask/cdap/examples/sportresults/UploadService.java]: include the message of the failure clause?

**Comment by chtyim** [comment]
Closing this PR first. Need to have some small adjustment to the API before updating the example again.

**Comment by anew** [comment]
L(mostly)GTM, with some minor comments.  
Do you think we should have an example that illustrates the use of transactions in onReceived()?

**Pull request closed without merging**